File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        int cnt[1001] = { 0 };
        vector<int> res;

        for (int n: nums1) cnt[n]++;

        for (int n: nums2) if (cnt[n] > 0) {
            res.push_back(n);
            cnt[n] = 0;
        }

        return res;
    }
};

auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 'c';
}();

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        int[] cnt = new int[1001];
        int[] res = new int[1001];

        for (int n: nums1) cnt[n]++;

        int idx = 0;
        for (int n: nums2) if (cnt[n] > 0) {
            res[idx++] = n;
            cnt[n] = 0;
        }

        return Arrays.copyOfRange(res, 0, idx);
    }
}